### DISCUSSION TITLE
Officially discourage 2019-09

### DISCUSSION THREAD

**Discussion by handrews** [json-schema-org/community#209] [Vision and direction]
2019-09 was not the latest draft long enough to see truly broad adoption before it was superseded by 2020-12.

2019-09 also has the `$recursive*` keywords, which neither do what they were supposed to do nor match the behavior of anything before or since.

And, of course, 2020-12 is compatible with OpenAPI 3.1, and 2019-09 is not.

Regardless of our [larger strategy around draft support](https://github.com/orgs/json-schema-org/discussions/192), I think we should make life easier for our community by explicitly discouraging the use or implementation of 2019-09 in favor of 2020-12.

At this point, there are too many options available, and by default people assume they should all be supported.  Reducing the number by guiding people to the most useful drafts will make life easier all around.  It would also be very easy to provide an official automatic 2019-09 to 2020-12 migration tool for anyone who did start using 2019-09.

**Comment by jviotti** [comment]
I personally agree with this, and this is aligned with the strategy I'm taking for JSON BinPack: support the latest version of the JSON Schema specification only. 

Regarding tooling, I was working on https://github.com/sourcemeta/alterschema/ for this very specific reason. I reckon I didn't spent much time on it lately, but various rules for transforming from 2019-09 to 2020-12 have been implemented already.

**Reply by handrews** [reply]
@jviotti Great to hear about the tooling!  I just want make clear that I'm not (in this discussion) arguing for only supporting the most recent.  That's a bigger decision than I think we're in a position to make.

But I feel that "Out of these two very similar versions, we should guide people to the one that doesn't have a pair of horribly broken keywords, and does have compatibility with OpenAPI, rather than the one with broken keywords and no compatibility" is a position that might get broad support regardless of how we treat past drafts in general.

**Reply by gregsdennis** [reply]
@jviotti your opinion would be welcome in the more general discussion that Henry linked to in the opening comment.  There we're talking about the overall policy for deprecation of spec versions.

**Comment by gregsdennis** [comment]
I think we can close this.  I hardly ever see it in use anymore, and when I do, I tell them to use 2020-12 instead or draft 7 if their implementation doesn't support 2020-12.
